docs/ingest-pipelines.asciidoc

```diff
@@ -0,0 +1,21 @@
+[[ingest-pipelines]]
+=== Parse data using ingest pipelines
+
+The APM integration loads default ingest node pipelines into {es}.
+These pipelines preprocess and enrich APM documents before indexing them.
+For example, a pipeline might define one processor that removes a field, and another that renames a field.
+
+Pipelines can be used to ensure data security by removing or obfuscating sensitive information.
+See <<apm-data-security,data security>> for an example.
+
+[float]
+[[view-edit-default-pipelines]]
+=== View or edit a default pipeline
+
+To view or edit a default pipelines in {kib},
```

Comment:
Let me know if I'm understanding this correctly:

1. Fleet adds ingest pipelines for the APM integration using index templates that include pipeline index settings. 
2. Elasticsearch uses the index pattern in these index templates to match pipelines to APM data streams.

As an example, the index template `traces-apm`, matches `traces-apm-*` data streams and assigns the `"traces-apm-7.16.0"` pipeline: 

<img width="606" alt="Screen Shot 2021-10-29 at 1 15 07 PM" src="https://user-images.githubusercontent.com/5618806/139496414-89ea3078-3dd8-4c26-9766-80a299dd9694.png">

3. This, Fleet managed, index template ⬆️  will be overwritten on reinstall/upgrade.

Assuming the above is correct, then I see three scenarios:

1. If a user edits the Fleet managed index template to point to a new pipeline, they need to:
a. ensure the package-provided pipeline is also called
b. know that the index template will be overwritten on install/upgrade

2. If a user creates a custom component template with a higher priority that points to a new pipeline, they need to:
a. ensure the package-provided pipeline is also called
b. _not sure what happens on upgrade here_

3. If a user edits the pipeline directly, like the `"traces-apm-7.16.0"` pipeline, they need to:
a. ensure the package-provided pipelines are still included
b. know that their edits will be overwritten on install/upgrade

Reply:
I had thought that the default pipeline was defined in the `@settings` component template, which would allow it to be overridden. That is not the case, so #2 isn't possible actually. I believe at the moment users would be limited to #1 and #3. We'll need to improve this. CC @simitt

Users shouldn't need to care about upgrades, but rather they should have some way of hooking their own custom pipeline into a predefined location in the Fleet-managed ingest pipeline.

Reply:
Perfect. Thanks, Andrew!

Reply:
started https://github.com/elastic/package-spec/issues/129 a while back; @ruflin do you have any concerns when moving the `default_pipeline` to the component-template instead of the index-template to make it user overridable?